devicetypes/smartthings/zwave-radiator-thermostat.src/zwave-radiator-thermostat.groovy

```diff
@@ -0,0 +1,254 @@
+	response(refresh())
+}
+
+def installed() {
+	initialize()
+}
+
+def updated() {
+	initialize()
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	} else {
+		log.warn "${device.displayName} - no-parsed event: ${description}"
+	}
+	log.debug "Parse returned: ${result}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		log.debug "SecurityMessageEncapsulation into: ${encapsulatedCommand}"
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "unable to extract secure command from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def value = cmd.batteryLevel == 255 ? 1 : cmd.batteryLevel
+	def map = [name: "battery", value: value, unit: "%"]
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatmodev2.ThermostatModeReport cmd) {
+	def map = [name: "thermostatMode", data:[supportedThermostatModes: thermostatSupportedModes]]
+	switch (cmd.mode) {
+		case 1:
+			map.value = "heat"
+			break
+		case 11:
+			map.value = "emergency heat"
+			break
+		case 0:
+			map.value = "off"
+			break
+	}
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointReport cmd) {
+	createEvent(name: "heatingSetpoint", value: convertTemperatureIfNeeded(cmd.scaledValue, 'C', cmd.precision), unit: temperatureScale)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	def map = [name: "temperature", value: convertTemperatureIfNeeded(cmd.scaledSensorValue, 'C', cmd.precision), unit: temperatureScale]
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	log.warn "Unhandled command: ${cmd}"
+	[:]
+}
+
+def setThermostatMode(String mode) {
+	def modeValue = 0
+	if (thermostatSupportedModes.contains(mode)) {
+		switch (mode) {
+			case "heat":
+				modeValue = 1
+				break
+			case "emergency heat":
+				modeValue = 11
+				break
+			case "off":
+				modeValue = 0
+				break
+		}
+	} else {
+		log.debug "Unsupported mode ${mode}"
+	}
+
+	[
+			secure(zwave.thermostatModeV2.thermostatModeSet(mode: modeValue)),
+			"delay 2000",
+			secure(zwave.thermostatModeV2.thermostatModeGet())
+	]
+}
+
+def heat() {
+	setThermostatMode("heat")
+}
+
+def emergencyHeat() {
+	setThermostatMode("emergency heat")
+}
+
+def off() {
+	setThermostatMode("off")
+}
+
+def setHeatingSetpoint(setpoint) {
+	setpoint = temperatureScale == 'C' ? setpoint : fahrenheitToCelsius(setpoint)
+	setpoint = Math.max(Math.min(setpoint, maxHeatingSetpointTemperature), minHeatingSetpointTemperature)
+	[
+			secure(zwave.thermostatSetpointV2.thermostatSetpointSet([precision: 1, scale: 0, scaledValue: setpoint, setpointType: 1, size: 2])),
```

Comment:
@greens  Given my comment above about the scale, does it make sense to convert to a scale that we know the device is using, or just force 'C'? If another device is added to this DTH and it is in "F" mode, if it receives a thermostat setpoint set with a value where the scale is "C" will it "do the right thing"? (I assume yes without pulling up the specs, but you know what assuming leads to...)

Reply:
I'd just leave a note near the fingerprints. Adding that conversion logic is a significant lift.

Reply:
@greens I've added a note, but I'm not sure if it is what you've though about